### PR TITLE
CoT: Do no use Github's committer data if it's 'web-flow'

### LINKED ISSUE #334: Committing with the Github's desktop app breaks Chain of Trust
Error spotted at https://treeherder.mozilla.org/#/jobs?repo=fenix&revision=4cd0d464a7d5e15cc85f29e0dc14fcd127cca4ea&selectedJob=242996267:

```python
2019-04-26T21:26:59 CRITICAL - signing:parent fzYG7Cy0RviQIijkGw8jvg: the runtime task doesn't match any rebuilt definition!
["[('change', 'payload.env.MOBILE_TRIGGERED_BY', ('web-flow', 'sblatz')),\n"
 " ('change',\n"
 "  'metadata.owner',\n"
 "  ('web-flow@users.noreply.github.com', 'sblatz@users.noreply.github.com'))]"]

```

https://api.github.com/repos/mozilla-mobile/fenix/commits/4cd0d464a7d5e15cc85f29e0dc14fcd127cca4ea returns:
```json
 "committer": {
    "login": "web-flow",
  }
```
which points to [this Github user](https://github.com/web-flow):

`git show --format=fuller 4cd0d464a7d5e15cc85f29e0dc14fcd127cca4ea` shows:
```
commit 4cd0d464a7d5e15cc85f29e0dc14fcd127cca4ea
Author:     Sawyer Blatz <sdblatz@gmail.com>
AuthorDate: Fri Apr 26 13:54:38 2019 -0700
Commit:     GitHub <noreply@github.com>
CommitDate: Fri Apr 26 13:54:38 2019 -0700
```

I confirmed with @sblatz that he used Github's desktop app to make this commit. That error happened on other commits of his. 

In https://github.com/mozilla-releng/scriptworker/pull/327, I changed CoT to use the committer data, instead of the author one because of a similar kind of breakage (#326).

We might want to change https://github.com/mozilla-releng/scriptworker/blob/8e97bbd83b9b578565ec57904c966dd6ae4ef0ae/scriptworker/cot/verify.py#L1201 to detect if the committer is `web-flow` and then fall back to the author data instead. We may run into another error if the person who merges the PR is different. Let's check that first.

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Fixes #334

### CODE REVIEW

File: scriptworker/test/test_cot_verify.py [hunk 1]
```diff
@@ -1353,11 +1353,19 @@ def test_verify_link_in_task_graph_exception(chain, decision_link, build_link, i
 
 
 @pytest.mark.asyncio
-async def test_populate_jsone_context_github_push(mocker, mobile_chain, mobile_github_push_link):
+@pytest.mark.parametrize('committer_login, author_login', (
+    ('some-user', 'some-user'),
+    ('some-user', 'some-other-user'),
+    ('web-flow', 'some-user'),
+)))
```
Comment (MihaiTabara) [inline scriptworker/test/test_cot_verify.py]: Nit: potential extra paranthesis here?

File: scriptworker/cot/verify.py [hunk 2]
```diff
@@ -1183,6 +1183,12 @@ def _get_additional_git_cron_jsone_context(decision_link):
         repo_owner, repo_name, context.config['github_oauth_token']
     ).get_commit(commit_hash)
 
+    committer_login = commit_data['committer']['login']
+    # https://github.com/mozilla-releng/scriptworker/issues/334: web-flow is the User used by
+    # GitHub to create some commits on github.com or the Github Desktop app. For sure, this user
+    # is not the one who triggered a push event. Let's fall back to the author login, instead.
+    sender_login = commit_data['author']['login'] if committer_login == 'web-flow' else committer_login
```
Comment (MihaiTabara) [inline scriptworker/cot/verify.py]: 👍 for explanatory comment.

File: scriptworker/test/test_cot_verify.py (hunk 1 above)
Comment (JohanLorenzo) [reply scriptworker/test/test_cot_verify.py]: That was it, thanks :)

**Comment by JohanLorenzo** [comment]
I don't feel super at ease to deploy this before I go on PTO. @MihaiTabara @mitchhentges, feel free to do, if the bug becomes too problematic.

**Comment by MihaiTabara** [comment]
This slipped my radar the past two days, I'll take care of this on Monday, 13th of May to avoid making changes to infra on a Friday.

**Pull request merged**